frontend/src/lib/WorkflowParser.ts

```diff
@@ -73,9 +72,29 @@ export default class WorkflowParser {
         statusMap.set(status[taskRunId]['pipelineTaskName'], status[taskRunId]);
     }
 
+    // Add When-condition tasks to conditionTasks Map if it depends on the result of the tasks in statusMap
+    const conditionTasks = new Map<string, any>();
```

Comment:
Could we use an array for this instead and just check conditionTasks.includes()? It doesn't seem that the value of the key, value pair ever gets used.

Reply:
Thanks @drewbutlerbb4 , My initial idea is want to keep the conditionTasks same as statusMap, to avoid anywhere else need the value of the key, but it's true it was not used finally, I modified it based on your comments, please check again. Thanks!